R/load_cell_graphs.R

```diff
@@ -214,6 +262,142 @@ LoadCellGraphs.CellGraphAssay <- function (
   return(object)
 }
 
+#' @param force Force load graph(s) if they are already loaded
+#' @param chunk_size An integer specifying a number of components to load
+#' in chunks.
+#' @param cl A cluster object created by makeCluster, or an integer
+#' to indicate number of child-processes (integer values are ignored
+#' on Windows) for parallel evaluations. See Details on performance
+#' in the documentation for \code{pbapply}. The default is NULL,
+#' which means that no parallelization is used.
+#'
+#' @rdname LoadCellGraphs
+#' @method LoadCellGraphs CellGraphAssay5
+#'
+#' @export
+#'
+LoadCellGraphs.CellGraphAssay5 <- function (
+  object,
+  cells = colnames(object),
+  load_as = c("bipartite", "Anode", "linegraph"),
+  add_marker_counts = TRUE,
+  force = FALSE,
+  chunk_size = 10,
+  cl = NULL,
+  verbose = TRUE,
+  ...
+) {
+
+  # Validate input parameters
+  stopifnot(
+    "'cells' must be a non-empty character vector of cell names" =
+      is.character(cells) &&
+      (length(cells) > 0)
+  )
+  stopifnot(
+    "'cells' must be present in 'object'" =
+      all(cells %in% colnames(object))
+  )
+  load_as <- match.arg(load_as, choices = c("bipartite", "Anode", "linegraph"))
+
+  # Check if cells are already loaded
+  loaded_graphs <- !sapply(slot(object, name = "cellgraphs")[cells], is.null)
+  if ((sum(!loaded_graphs) == 0) & !force) {
+    if (verbose && check_global_verbosity())
+      cli_alert_info("All cells are alredy loaded. Returning CellGraphAssay unmodified.")
+    return(object)
+  } else {
+    if (force) {
+      slot(object, "cellgraphs")[cells] <- rep(list(NULL), length(cells)) %>% setNames(nm = cells)
+      loaded_graphs <- !sapply(slot(object, name = "cellgraphs")[cells], is.null)
+    }
+    cells_to_load <- setdiff(cells, cells[loaded_graphs])
+    if (verbose && check_global_verbosity() & (length(cells_to_load) < length(cells)))
+      cli_alert_info(glue("{length(cells) - length(cells_to_load)} CellGraphs loaded.",
+                          " Loading remaining {length(cells_to_load)} CellGraphs."))
+  }
+
+  # Find where components are located
+  fs_map_unnested_filtered <- slot(object, name = "fs_map") %>%
```

Comment:
This is the new slot provided in the CellGraphAssay5 class. It's a tibble with 3 columns:

- id_map : a list of tibbles
   - each tibble has a current_id column with the current component IDs and a original_id column with the original MPX IDs
- sample : an integer for each sample (you get multiple samples here if the object is merged)
- pxl_file : paths to the pxl files

Reply:
This is all information we need to know in order to map a set of component IDs to a specific pxl file. When merging data, we only need to merge these tables instead of having to go into the edgelists and fiddle around.